src/debug/FuncTaskProvider.ts

```diff
@@ -73,20 +73,50 @@ export class FuncTaskProvider implements TaskProvider {
         return result;
     }
 
-    public async resolveTask(_task: Task, _token?: CancellationToken | undefined): Promise<Task | undefined> {
-        await callWithTelemetryAndErrorHandling('resolveTask', async (context: IActionContext) => {
+    public async resolveTask(task: Task, _token?: CancellationToken | undefined): Promise<Task | undefined> {
+        return await callWithTelemetryAndErrorHandling('resolveTask', async (context: IActionContext) => {
             context.telemetry.properties.isActivationEvent = 'true';
             context.errorHandling.suppressDisplay = true;
             context.telemetry.suppressIfSuccessful = true;
+
+            // tslint:disable-next-line: no-unsafe-any
+            const command: string | undefined = task.definition.command;
+            if (command && task.scope !== undefined && task.scope !== TaskScope.Global && task.scope !== TaskScope.Workspace) {
```

Comment:
Would it not be an error if we think there's a task to resolve, but it's undefined?

Reply:
We suppress errors (see above) so doesn't really matter if we throw an error here or not. resolveTask gets called a lot by VS Code and so we don't want to show errors when people might not even be trying to run a task from us